services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -676,24 +681,51 @@ async def _get_size_and_num_files(
 
         return parse_obj_as(ByteSize, total_size), total_num_s3_objects
 
-    async def search_files_starting_with(
-        self, user_id: UserID, prefix: str
+    async def search_read_access_files(
+        self, user_id: UserID, file_id_prefix: str, sha256_checksum: SHA256Str | None
+    ):
+        async with self.engine.acquire() as conn:
+            can_read_projects_ids = await get_readable_project_ids(conn, user_id)
+        return await self._search_files(
+            user_id=user_id,
+            project_ids=can_read_projects_ids,
+            file_id_prefix=file_id_prefix,
+            sha256_checksum=sha256_checksum,
+        )
+
+    async def search_owned_files(
+        self, user_id: UserID, file_id_prefix: str, sha256_checksum: SHA256Str | None
+    ):
+        return await self._search_files(
+            user_id=user_id,
+            project_ids=[],
+            file_id_prefix=file_id_prefix,
+            sha256_checksum=sha256_checksum,
+        )
```

Comment:
These could have easily been defined as one function. It will also look better when calling it in the API handler.

Reply:
I agree. I preferred to do it like this to really make 100% clear among which files are searched. This was not clear to me before and that caused a lot of confusion for me. I think like this it is much more clear, but of course also a bit more verbose...

Reply:
OK let's keep the split.